### ISSUE TITLE
Unable to compile 0.6.0 or 0.7.0 on Ubuntu 16.10

### ISSUE THREAD

**Issue by arnvid** [vovoid/vsxu#180]
Trying to compile up vsxu for windows on Ubuntu 16.10 and I end up with an interesting 

Building win64 gives:
/home/arnvid/vsxu-dev/build/win64/dependencies-install
-- The C compiler identification is GNU 6.2.0
-- The CXX compiler identification is GNU 6.2.0
-- Check for working C compiler: /usr/bin/x86_64-w64-mingw32-gcc
-- Check for working C compiler: /usr/bin/x86_64-w64-mingw32-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/x86_64-w64-mingw32-g++
-- Check for working CXX compiler: /usr/bin/x86_64-w64-mingw32-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
---------------------------------------------------------------------
                     Configuring Vovoid VSXu
=====================================================================
configuring            vsx_engine_graphics
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found OpenGL: opengl32
configuring vsxu module            bitmap.generators
bitmap generators: /home/arnvid/vsxu-dev/vsxu/plugins/src/bitmap.generators/main_bitmap.generators.cpp /home/arnvid/vsxu-dev/vsxu/plugins/src/bitmap.generators/_configuration.h;/home/arnvid/vsxu-dev/vsxu/plugins/src/bitmap.generators/module_bitmap_blob.h;/home/arnvid/vsxu-dev/vsxu/plugins/src/bitmap.generators/module_bitmap_concentric_circles.h;/home/arnvid/vsxu-dev/vsxu/plugins/src/bitmap.generators/module_bitmap_perlin_noise.h;/home/arnvid/vsxu-dev/vsxu/plugins/src/bitmap.generators/module_bitmap_plasma.h;/home/arnvid/vsxu-dev/vsxu/plugins/src/bitmap.generators/module_bitmap_subplasma.h
configuring vsxu module            bitmap.loaders
configuring vsxu module            bitmap.modifiers
configuring vsxu module            _deprecated
configuring vsxu module            math
configuring vsxu module            math.oscillators
configuring vsxu module            mesh.generators
configuring vsxu module            mesh.importers
configuring vsxu module            mesh.modifiers
configuring vsxu module            mesh.render
configuring vsxu module            outputs.screen_opengl
configuring vsxu module            particlesystem.generators
configuring vsxu module            particlesystem.modifiers
configuring vsxu module            particlesystem.render
configuring vsxu module            render.basic
configuring vsxu module            render.glsl
configuring vsxu module            render.gravity_lines
configuring vsxu module            render.opengl
configuring vsxu module            render.text
configuring vsxu module            selectors
configuring vsxu module            sound.rtaudio
winmmdsound
audio libraries:

winmmdsound
configuring vsxu module            string
configuring vsxu module            system
configuring vsxu module            texture
configuring            vsx_application
configuring            vsx_common
-- Looking for dlopen
-- Looking for dlopen - not found
configuring            vsx_compression
configuring            vsx_engine
configuring            vsx_widget
configuring            vsxu_artiste
configuring            vsxu_player
configuring            vsxu_profiler
configuring            vsxbt
configuring            vsxz
configuring            obj2vxm
-- Configuring done
-- Generating done
-- Build files have been written to: /home/arnvid/vsxu-dev/build/win64/build
/home/arnvid/vsxu-dev/build/win64/dependencies-install/include
Scanning dependencies of target vsx_compression
[  0%] Building CXX object lib/compression/CMakeFiles/vsx_compression.dir/src/vsx_compression_lzma.cpp.obj
In file included from /home/arnvid/vsxu-dev/vsxu/lib/compression/include/vsx_compression_lzma.h:3:0,
                 from /home/arnvid/vsxu-dev/vsxu/lib/compression/src/vsx_compression_lzma.cpp:4:
/home/arnvid/vsxu-dev/vsxu/lib/common/include/container/vsx_ma_vector.h: In member function ‘void vsx_ma_vector<T>::allocate_bytes(size_t)’:
/home/arnvid/vsxu-dev/vsxu/lib/common/include/container/vsx_ma_vector.h:176:35: error: there are no arguments to ‘vsx_aligned_malloc’ that depend on a template parameter, so a declaration of ‘vsx_aligned_malloc’ must be available [-fpermissive]
     A = (T*)vsx_aligned_malloc( b );
                                   ^
/home/arnvid/vsxu-dev/vsxu/lib/common/include/container/vsx_ma_vector.h:176:35: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)
/home/arnvid/vsxu-dev/vsxu/lib/common/include/container/vsx_ma_vector.h: In member function ‘void vsx_ma_vector<T>::allocate(size_t)’:
/home/arnvid/vsxu-dev/vsxu/lib/common/include/container/vsx_ma_vector.h:194:74: error: there are no arguments to ‘vsx_aligned_malloc’ that depend on a template parameter, so a declaration of ‘vsx_aligned_malloc’ must be available [-fpermissive]
         A = (T*)vsx_aligned_malloc(sizeof(T)*(index+allocation_increment));
                                                                          ^
/home/arnvid/vsxu-dev/vsxu/lib/common/include/container/vsx_ma_vector.h: In instantiation of ‘vsx_ma_vector<T>::~vsx_ma_vector() [with T = unsigned char]’:
/home/arnvid/vsxu-dev/vsxu/lib/compression/src/vsx_compression_lzma.cpp:11:32:   required from here
/home/arnvid/vsxu-dev/vsxu/lib/common/include/container/vsx_ma_vector.h:290:21: error: ‘vsx_aligned_free’ was not declared in this scope
     vsx_aligned_free(A);
     ~~~~~~~~~~~~~~~~^~~
/home/arnvid/vsxu-dev/vsxu/lib/common/include/container/vsx_ma_vector.h: In instantiation of ‘void vsx_ma_vector<T>::allocate(size_t) [with T = unsigned char; size_t = long long unsigned int]’:
/home/arnvid/vsxu-dev/vsxu/lib/compression/src/vsx_compression_lzma.cpp:14:47:   required from here
/home/arnvid/vsxu-dev/vsxu/lib/common/include/container/vsx_ma_vector.h:191:36: error: ‘vsx_aligned_realloc’ was not declared in this scope
         A = (T*)vsx_aligned_realloc(A,sizeof(T)*allocated);
                 ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/home/arnvid/vsxu-dev/vsxu/lib/common/include/container/vsx_ma_vector.h:194:35: error: ‘vsx_aligned_malloc’ was not declared in this scope
         A = (T*)vsx_aligned_malloc(sizeof(T)*(index+allocation_increment));
                 ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
lib/compression/CMakeFiles/vsx_compression.dir/build.make:63: recipe for target 'lib/compression/CMakeFiles/vsx_compression.dir/src/vsx_compression_lzma.cpp.obj' failed
make[2]: *** [lib/compression/CMakeFiles/vsx_compression.dir/src/vsx_compression_lzma.cpp.obj] Error 1
CMakeFiles/Makefile2:2327: recipe for target 'lib/compression/CMakeFiles/vsx_compression.dir/all' failed
make[1]: *** [lib/compression/CMakeFiles/vsx_compression.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2

Buiding win32 gives this error;

Scanning dependencies of target test_cpu_info
Scanning dependencies of target vsx_compression
Scanning dependencies of target test_vsx_profiler
Scanning dependencies of target vsx_application
[  1%] Building CXX object programs/profiler/test/CMakeFiles/test_vsx_profiler.dir/test_vsx_profiler.cpp.obj
[  1%] Building CXX object lib/common/test/CMakeFiles/test_cpu_info.dir/test_cpu_info.cpp.obj
[  1%] Building CXX object lib/application/CMakeFiles/vsx_application.dir/src/vsx_application_manager.cpp.obj
[  2%] Building CXX object lib/application/CMakeFiles/vsx_application.dir/src/vsx_application_control.cpp.obj
[  3%] Building CXX object lib/application/CMakeFiles/vsx_application.dir/src/sdl/vsx_application_run.cpp.obj
[  3%] Building CXX object lib/application/CMakeFiles/vsx_application.dir/src/sdl/vsx_application_input_mouse_control.cpp.obj
In file included from /home/arnvid/vsxu-dev/vsxu/lib/application/src/sdl/vsx_application_run.cpp:1:0:
/home/arnvid/vsxu-dev/vsxu/lib/engine_graphics/include/vsx_gl_global.h:28:27: fatal error: GL/glew.h: No such file or directory
       #include <GL/glew.h>
                           ^
compilation terminated.

**Comment by jaw** [comment]
Hi,

good point. Haven't noticed since we've basically moved to compiling windows on windows.

Do you have a windows machine available?

**Comment by arnvid** [comment]
yes - ofcoz :) I have a few ;) - I was building the 0.7.0 since the 0.5.1 keeps eating 32GB of memory when I am streaming dj sets on twitch with the result of a complete systems meltdown

**Comment by jaw** [comment]
yea, fixed most leaks, 0.6.0 leaks like 500MB per 24 hours (not really worth fixing).

0.6.0 feature development has ceased, we're just tweaking some small things and finalizing the release.
If you want stability, stay away from 0.7.0, it's the new development branch :)

Also see the INSTALL file in 0.6.0, I updated it with what to install to build on windows.

And, here is 0.6.0 installer: 
https://www.dropbox.com/s/78zh16dsb3oe4xo/VSXu_0.6.0_amd64.exe?dl=0

It's still not 100% (mainly the launcher), so keep that in mind.

**Comment by arnvid** [comment]
I don't mind testing 0.7.0  - gonna try to compile it and see how it runs.
Thanks for the 0.6.0 installer.
